lisa/base_tools/service.py

```diff
@@ -235,19 +235,29 @@ def enable_service(self, name: str) -> None:
         pass
 
     def restart_service(self, name: str, ignore_exit_code: int = 0) -> None:
-        self.node.tools[PowerShell].run_cmdlet(
-            f"Restart-service {name}",
-            force_run=True,
-        )
+        try:
+            self.node.tools[PowerShell].run_cmdlet(
+                f"Restart-service {name}",
+                force_run=True,
+            )
+        except LisaException as identifier:
+            if "Cannot find any service with service name" in str(identifier):
+                self._log.debug(f"service '{name}' does not exist")
+                return
+            raise identifier
         self.wait_for_service_start(name)
 
     def stop_service(self, name: str) -> None:
-        self.node.tools[PowerShell].run_cmdlet(
-            f"Stop-Service {name} -Force",
-            force_run=True,
-            output_json=True,
-            fail_on_error=False,
-        )
+        try:
+            self.node.tools[PowerShell].run_cmdlet(
+                f"Stop-Service {name} -Force",
+                force_run=True,
+            )
+        except LisaException as identifier:
+            if "Cannot find any service with service name" in str(identifier):
```

Comment:
With this error message, when a non-exist service will be stop or restarted in a normal state? if the service doesn't exist, it should fail earlier, instead of stop/start it without errors.

Reply:
Adding an extra call to check_service_exists will add latency. Since restart/stop can capture it as part of the reason for failure. I think we can keep this message. 

I added extra messages to capture reasons for not stopping a service. One of the reasons for not able to stop is dependent services.

Reply:
The up-layer logic should make sure start/stop are not called, if the service doesn't exist, instead of eating the exception here. It may hide the problem, and fails on other steps and confusing. So it doesn't need to eat any exception here, just let the code raise as is.

When the service is used, if the service doesn't exist, the test case level should fix it or skip the test case, instead of continuing to start/stop service or any other logic.